From 650aa62ebb77764338663e8ca5b11430e3eea970 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 9 Sep 2024 14:39:39 +0200
Subject: [PATCH 1078/1695] dts: stm32: describe vrefbuf supply for stm32mp23

Describe PWR VDDA18ADC as VREFBUF supply for STM32MP23.
The goal is to guaranty that VDDA18ADC SV is set before enabling VREFBUF.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1f04a744266a8641748c501ff367971de32fc8ad dts: stm32: describe vrefbuf supply for stm32mp23]
Change-Id: I7a74cd8967bbe4b39f4400ead1667c2620ec006e
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403074
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 562b43e5a..3962420ca 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -697,6 +697,7 @@
 			regulator-min-microvolt = <1210000>;
 			regulator-max-microvolt = <1500000>;
 			clocks = <&rcc CK_BUS_VREF>;
+			vdda-supply = <&vdda18adc>;
 			status = "disabled";
 		};
 
-- 
2.47.2

